### PR TITLE
Material: Fix Compile() when NeedsCompile() = false

### PR DESCRIPTION
**Pull request by CyaNinix**
When a material property is updated on a material that doesn't have this property (which of course is a mistake), the material becomes stuck and cannot compile. 

As `Material::CanCompile` will return false (because its srg will not be queued for compilation), the material will simply try to compile every tick forever.

A simple fix is to check if the material compilation is required.

### CODE REVIEW

**Review by cgalvan — CHANGES_REQUESTED** [review]

File: Gems/Atom/RPI/Code/Source/RPI.Public/Material/Material.cpp [hunk 1]
```diff
@@ -340,7 +340,10 @@ namespace AZ
         {
             AZ_PROFILE_FUNCTION(RPI);
 
-            if (NeedsCompile() && CanCompile())
+            if (!NeedsCompile())
+                return true;
```
[the remark this answers is not in the document]
Comment (cgalvan) [reply Gems/Atom/RPI/Code/Source/RPI.Public/Material/Material.cpp]: Wow you are fast! Kicking off AR

**Comment by cgalvan** [comment]
@CyaNinix This unit test failed with this change:
```
[2022-06-13T16:57:09.946Z] [ RUN      ] MaterialTests.TestSetPropertyValueWhenValueIsUnchanged
[2022-06-13T16:57:09.946Z] D:/workspace/o3de/Gems/Atom/RPI/Code/Tests/Material/MaterialTests.cpp(357): error: Value of: material->Compile()
[2022-06-13T16:57:09.946Z]   Actual: true
[2022-06-13T16:57:09.946Z] Expected: false
[2022-06-13T16:57:09.946Z] [  FAILED  ] MaterialTests.TestSetPropertyValueWhenValueIsUnchanged (14 ms)
```

**Comment by CyaNinix** [comment]
> This unit test failed with this change:

Indeed, but then with this change the behavior of Compile() changes when a property value is unchanged. Should I fix the test?

**Comment by cgalvan** [comment]
> > This unit test failed with this change:
> 
> Indeed, but then with this change the behavior of Compile() changes when a property value is unchanged. Should I fix the test?

Yeah I think in this case since we changed the exact behavior the test was trying to validate it is valid to update the test.

**Review by cgalvan — APPROVED** [review]

**Review by gadams3 — APPROVED** [review]

**Pull request merged**